svir/processing_provider/spatial_join_summary_style.py

```diff
@@ -114,321 +83,33 @@ def tr(self, string):
-                    field_type = field_types[i]
-                    if field_type == 'numeric':
-                        stat = QgsStatisticalSummary()
-                        for v in attribute_values:
-                            stat.addVariant(v)
-                        stat.finalize()
-                        for s in numeric_fields:
-                            if s[0] in summaries:
-                                val = getattr(stat, s[2])()
-                                attrs.append(
-                                    val if not math.isnan(val) else NULL)
-                    elif field_type == 'datetime':
-                        stat = QgsDateTimeStatisticalSummary()
-                        stat.calculate(attribute_values)
-                        for s in datetime_fields:
-                            if s[0] in summaries:
-                                if s[0] == 'filled':
-                                    attrs.append(
-                                        stat.count() - stat.countMissing())
-                                elif s[0] == 'min':
-                                    attrs.append(stat.statistic(
-                                        QgsDateTimeStatisticalSummary.Min))
-                                elif s[0] == 'max':
-                                    attrs.append(stat.statistic(
-                                        QgsDateTimeStatisticalSummary.Max))
-                                else:
-                                    attrs.append(getattr(stat, s[2])())
-                    else:
-                        stat = QgsStringStatisticalSummary()
-                        for v in attribute_values:
-                            if v == NULL:
-                                stat.addString('')
-                            else:
-                                stat.addString(str(v))
-                        stat.finalize()
-                        for s in string_fields:
-                            if s[0] in summaries:
-                                if s[0] == 'filled':
-                                    attrs.append(
-                                        stat.count() - stat.countMissing())
-                                else:
-                                    attrs.append(getattr(stat, s[2])())
-
-                f.setAttributes(attrs)
-                sink.addFeature(f, QgsFeatureSink.FastInsert)
+    def postProcessAlgorithm(self, context, feedback):
 
-        return {self.OUTPUT: self.dest_id,
-                self.ADDED_FIELDS: self.fields_to_join}
+        input_layer = QgsProcessingUtils.mapLayerFromString(
+            self.parameters[self.INPUT], context)
 
-    def postProcessAlgorithm(self, context, feedback):
         processed_layer = QgsProcessingUtils.mapLayerFromString(
             self.dest_id, context)
-        added_fieldnames = [field.name() for field in self.fields_to_join]
+
+        added_fieldnames = set(processed_layer.fields().names()) - set(
```

Comment:
Please check if this does not preserve the order of fields. Perhaps it should be slightly better to do a little loop to preserve the order.

Reply:
done